go/vt/tableacl/ldapacl/ldapacl.go

```diff
@@ -0,0 +1,151 @@
+package ldapacl
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	log "github.com/golang/glog"
+
+	"github.com/youtube/vitess/go/ldap"
+	"github.com/youtube/vitess/go/vt/tableacl"
+	"github.com/youtube/vitess/go/vt/tableacl/acl"
+	ldapv2 "gopkg.in/ldap.v2"
+)
+
+var (
+	ldapACLConfigFile   = flag.String("mysql_ldap_acl_config_file", "", "JSON File from which to read LDAP server config.")
+	ldapACLConfigString = flag.String("mysql_ldap_acl_config_string", "", "JSON representation of LDAP server config.")
+)
+
+var queryLdap func(string) ([]string, error)
+
+// ACLLdapServerConfig holds the ACL-specific configuration for an LDAP connection
+// It is recommended that User be read-only
+type ACLLdapServerConfig struct {
+	ldap.ServerConfig
+	User     string
+	Password string
+	Query    string
+}
+
+func init() {
+	if *ldapACLConfigFile == "" && *ldapACLConfigString == "" {
+		log.Infof("Not configuring Ldap ACLs because mysql_ldap_acl_config_file and mysql_ldap_acl_config_string are empty")
+		return
+	}
+	if *ldapACLConfigFile != "" && *ldapACLConfigString != "" {
+		log.Infof("Both mysql_ldap_acl_config_file and mysql_ldap_acl_config_string are non-empty, can only use one.")
+		return
+	}
+	config := &ACLLdapServerConfig{ldap.ServerConfig{}, "", "", ""}
+
+	data := []byte(*ldapACLConfigString)
+	if *ldapACLConfigFile != "" {
+		var err error
+		data, err = ioutil.ReadFile(*ldapACLConfigFile)
+		if err != nil {
+			log.Fatalf("Failed to read mysql_ldap_acl_config_file: %v", err)
+		}
+	}
+	if err := json.Unmarshal(data, config); err != nil {
+		log.Fatalf("Error parsing AuthServerLdap config: %v", err)
+	}
+	queryLdap = makeQueryClosure(config)
+	tableacl.Register("ldapacl", &Factory{})
+	tableacl.SetDefaultACL("ldapacl")
+	go refreshCachedUsers()
```

Comment:
We  would probably use a time.Timer and regularly refresh the cache.

The alternative is to store the time the ldap query was done, and if you try to access a value in the cache that is older than a threshold, you re-ask for it. You probably don't even need to clean up old values in the cache then.